src/plugins/data/server/search/opensearch_search/opensearch_search_strategy.test.ts

```diff
@@ -131,20 +150,96 @@ describe('OpenSearch search strategy', () => {
     expect(response).toHaveProperty('rawResponse');
   });
 
-  it('dataSource enabled, send request with dataSourceId get data source client', async () => {
-    const opensearchSearch = await opensearchSearchStrategyProvider(mockConfig$, mockLogger);
+  it('dataSource enabled, config host exist, send request with dataSourceId should get data source client', async () => {
+    const mockOpenSearchServiceSetup = opensearchServiceMock.createSetup();
+    mockOpenSearchServiceSetup.legacy.client = {
+      callAsInternalUser: jest.fn(),
+      asScoped: jest.fn(),
+      config: {
+        hosts: ['some host'],
+      },
+    };
+
+    const opensearchSearch = opensearchSearchStrategyProvider(
+      mockConfig$,
+      mockLogger,
+      mockSearchUsage,
+      mockDataSourcePluginSetupWithDataSourceEnabled,
+      mockOpenSearchServiceSetup
+    );
 
     await opensearchSearch.search(
       (mockDataSourceEnabledContext as unknown) as RequestHandlerContext,
       {
         dataSourceId,
       }
     );
+
+    expect(mockDataSourceApiCaller).toBeCalled();
+    expect(mockOpenSearchApiCaller).not.toBeCalled();
+  });
+
+  it('dataSource enabled, config host exist, send request without dataSourceId should get default client', async () => {
```

Comment:
One change in your RP is the error code 400 instead of 500. Can we also have a test to cover this as well? Thanks

Reply:
Test case updated here:
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5882/commits/456ee1324908ba7a352db260d5e966a2d070a667#diff-96950b20277c55c90d7592913fcc77ed6bac02a71701c784f2ac3b03c31d25d6R250

Thanks for the comment.